scripts/local-dev/setup-rbe.sh

```diff
@@ -2,69 +2,74 @@
-if [[ -f $json_key_filepath ]]; then
-  echo "Angular Local Dev key already exists, reusing this key.";
-else
-  # Confirm the user is already logged into gcloud, if they aren't
-  # attempt to login
-  echo "Checking gcloud login state.";
-  gcloud auth print-identity-token &> /dev/null;
-  if [[ $? -ne 0 ]]; then
-    echo "Not currently logged into gcloud. Starting gcloud login now.";
-    gcloud auth login;
-    if [[ $? -ne 0 ]]; then
-      echo "gcloud login failed. Aborting.";
-      exit 2;
-    fi
-  fi
-  gcloud iam service-accounts keys create $json_key_filepath \
-    --iam-account angular-local-dev@internal-200822.iam.gserviceaccount.com \
-    --quiet --project internal-200822;
+# Confirm the user is already logged into gcloud, if they aren't
+# attempt to login
+echo "Checking gcloud login state"
+gcloud auth application-default print-access-token &> /dev/null
+if [[ $? -ne 0 ]]; then
+  echo "Not currently logged into gcloud. Starting gcloud login now"
+  gcloud auth application-default login
   if [[ $? -ne 0 ]]; then
-    echo "Downloading service account key failed. Aborting.";
-    exit 2;
+    echo "gcloud login failed. Aborting"
+    exit 2
   fi
 fi
+access_token=$(gcloud auth application-default print-access-token)
+current_account=$(curl -s https://www.googleapis.com/oauth2/v1/tokeninfo?access_token=$access_token | jq -r '.email')
+if [[ ! $current_account =~ (angular\.io$)|(google\.com$) ]]; then
+  echo "Currently an angular.io or google.com account must be used for remote Bazel usage"
+  echo "Please login instead using the correct account with the following command, then rerun"
+  echo "  gcloud auth application-default login"
+  exit 3
+fi
+echo "Logged in as $current_account";
 
 # The full path to the .bazelrc.user file
-bazelrc_user_filepath="$project_directory/.bazelrc.user";
+bazelrc_user_filepath="$project_directory/.bazelrc.user"
 # Create the bazelrc.user file, echo the config flags into the file.
-touch $bazelrc_user_filepath;
-echo "build --config=remote-http-caching" >> $bazelrc_user_filepath;
-echo "build --google_credentials=$json_key_filepath" >> $bazelrc_user_filepath;
+touch $bazelrc_user_filepath
+
+# Prompts to add a flag to the .bazelrc.user file if its not already in place
+function add_flag() {
+  flag=$1
+  read -p "  Add $flag flag? [Y/y]"
+  if [[ $REPLY =~ ^[Yy]$ ]]; then
+    if [[ ! $(grep "^$flag$" $bazelrc_user_filepath) ]]; then
+      echo "$flag" >> $bazelrc_user_filepath
+      echo "Added $flag to .bazelrc.user"
+    else
+      echo "$flag already in .bazelrc.user"
+    fi
+  fi
+  echo
+}
+
+# Add extra line space before config setup.
+echo
+# Remote HTTP Caching
+echo "The ${bold}remote-http-caching${normal} flag enables uploading build results to the http cache,"
+echo "but not does enable remote builds"
+add_flag "build --config=remote-http-caching"
+
+# Remote builds
+echo "The ${bold}remote${normal} flag enables RBE, builds occurs remotely when possible and caching"
```

Comment:
Nit suggestion (mostly because `occurs` should be `occur` I guess)

```suggestion
echo "The ${bold}remote${normal} flag enables RBE, builds run remotely when possible and caching"
```